fw/Core/Hitcon/Logic/XBoardLogic.h

```diff
@@ -0,0 +1,72 @@
+#ifndef HITCON_LOGIC_XBOARD_LOGIC_H_
+#define HITCON_LOGIC_XBOARD_LOGIC_H_
+
+#include <Util/callback.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#include "Service/Sched/Scheduler.h"
+#include "Service/XBoardService.h"
+#include "usart.h"
+
+namespace hitcon {
+namespace service {
+namespace xboard {
+
+struct PacketCallbackArg {
+	uint8_t *data;
+	uint8_t len;
+};
+
+enum UsartConnectState { Init, Connect, Disconnect };
+
+constexpr size_t RX_BUF_SZ = 512;
```

Comment:
Might want to lower this to reduce memory usage, we're slightly short on memory.